Template.hs missing from Hackage tarball (sdist)

It is impossible to build the kdl-hs test suite from Hackage at the moment since the preprocessor is missing source files:

```
Building executable 'kdl-hs-test-decoder' for kdl-hs-1.1.0...
[1 of 1] Compiling Main             ( test/kdl-hs-test-decoder.hs, dist/build/kdl-hs-test-decoder/kdl-hs-test-decoder-tmp/Main.o )
[2 of 2] Linking dist/build/kdl-hs-test-decoder/kdl-hs-test-decoder
Preprocessing test suite 'kdl-tests' for kdl-hs-1.1.0...
Building test suite 'kdl-tests' for kdl-hs-1.1.0...
test/KDL/Decoder/SharedSpec/Arrow.hs:6:2: error:
     fatal error: ./Template.hs: No such file or directory

compilation terminated.
test/KDL/Decoder/SharedSpec/Monad.hs:6:2: error:
     fatal error: ./Template.hs: No such file or directory

compilation terminated.
test/KDL/Decoder/SharedSpec/Arrow.hs:6:2: error:
     fatal error: ./Template.hs: No such file or directory

compilation terminated.
test/KDL/Decoder/SharedSpec/Monad.hs:6:2: error:
     fatal error: ./Template.hs: No such file or directory

compilation terminated.
test/KDL/Decoder/SharedSpec/Arrow.hs:1:1: error:
    `cc' failed in phase `Haskell C pre-processor'. (Exit code: 1)
  |
1 | {-# LANGUAGE CPP #-}
  | ^

test/KDL/Decoder/SharedSpec/Monad.hs:1:1: error:
    `cc' failed in phase `Haskell C pre-processor'. (Exit code: 1)
  |
1 | {-# LANGUAGE CPP #-}
  | ^
```

At least `Template.hs` needs to be added to `extra-source-files`.

--- a/kdl-hs.cabal
+++ b/kdl-hs.cabal
@@ -15,6 +15,7 @@ build-type: Simple
 extra-source-files:
   README.md
   CHANGELOG.md
+  test/KDL/Decoder/SharedSpec/Template.hs
   test/KDL/__snapshots__/DecoderSpec.snap.md
   test/KDL/__snapshots__/ParserSpec.snap.md
   test/KDL/__snapshots__/RenderSpec.snap.md

--- a/kdl-hs.cabal
+++ b/kdl-hs.cabal
@@ -19,7 +19,6 @@ extra-source-files:
   test/KDL/__snapshots__/DecoderSpec.snap.md
   test/KDL/__snapshots__/ParserSpec.snap.md
   test/KDL/__snapshots__/RenderSpec.snap.md
-  tools/kdl-test
 
 source-repository head
   type: git

--- a/test/KDL/ParserSpec.hs
+++ b/test/KDL/ParserSpec.hs
@@ -9,7 +9,7 @@ import Data.Text.IO qualified as Text
 import KDL qualified
 import Skeletest
 import Skeletest.Predicate qualified as P
-import System.Directory (findExecutable, listDirectory)
+import System.Directory (doesFileExist, findExecutable, listDirectory)
 import System.FilePath (takeExtension, (</>))
 import System.IO.Temp (withSystemTempDirectory)
 import System.IO.Unsafe (unsafePerformIO)
@@ -46,7 +46,7 @@ spec = do
         -- tested in `parse`
         actual `shouldBe` expected
 
-  (if dotSlashInstalled then id else skip "dotslash not installed") . describe "kdl-test examples" $ do
+  (if runKdlTest then id else skip "cannot run kdl-test") . describe "kdl-test examples" $ do
     it "decodes correctly" $ do
       decoder <- findExecutable "kdl-hs-test-decoder" >>= maybe (error "Could not find kdl-hs-test-decoder") pure
       callProcess "tools/kdl-test" ["run", "--decoder", decoder]
@@ -61,5 +61,9 @@ spec = do
           content <- Text.readFile (dir </> "valid" </> file)
           (fmap KDL.render . KDL.parse) content `shouldBe` Right content
 
-dotSlashInstalled :: Bool
-dotSlashInstalled = unsafePerformIO $ isJust <$> findExecutable "dotslash"
+runKdlTest :: Bool
+runKdlTest = unsafePerformIO $ do
+  -- Not included in sdist because sdist doesn't preserve executable perms
+  scriptExists <- doesFileExist "tools/kdl-test"
+  dotslashInstalled <- isJust <$> findExecutable "dotslash"
+  pure $ scriptExists && dotslashInstalled

